sql/src/main/java/io/crate/executor/transport/task/elasticsearch/EsJobContextTask.java

```diff
@@ -61,27 +61,34 @@ protected void createContext(String operationName,
                                  TransportAction transportAction,
                                  @Nullable FlatProjectorChain projectorChain) {
         ESJobContext esJobContext = new ESJobContext(executionPhaseId, operationName,
-                requests, listeners, results, transportAction, projectorChain);
+            requests, listeners, results, transportAction, projectorChain);
         JobExecutionContext.Builder contextBuilder = jobContextService.newBuilder(jobId());
         contextBuilder.addSubContext(esJobContext);
         context = jobContextService.createContext(contextBuilder);
     }
 
     @Override
-    final public void start() {
+    public ListenableFuture<TaskResult> execute() {
         assert context != null : "Context must be created first";
+        SettableFuture<TaskResult> result = results.get(0);
         try {
             context.start();
         } catch (Throwable throwable) {
-            for (SettableFuture<TaskResult> result : results) {
-                result.setException(throwable);
-            }
+            result.setException(throwable);
         }
+        return result;
     }
 
     @Override
-    final public List<? extends ListenableFuture<TaskResult>> result() {
-        assert results.size() > 0 : "Result list is empty, sub-class muss add at least one";
+    public List<? extends ListenableFuture<TaskResult>> executeBulk() {
+        assert context != null : "Context must be created first";
```

Comment:
the only place that I could check it is used is [here](https://github.com/crate/crate/blob/546e4c15c23c95b31805331865b6fbe3febe9917/sql/src/test/java/io/crate/integrationtests/TransportSQLActionTest.java#L1254)